File: 7-kyu/simple-fun-#50--array-conversion/javascript/solution.js
const arrayConversion = (arr) => {
  let array = arr.map(elem => elem);
  let iter = 0;
  while (array.length > 1) {
    let temp = [];
    if (iter % 2 === 0) {
      for (let i = 0; i < array.length - 1; i += 2) {
        temp.push(array[i] + array[i + 1]);
      }
      array = temp;
    } else {
      for (let i = 0; i < array.length- 1; i += 2) {
        temp.push(array[i] * array[i + 1])
      }
      array = temp;
    }
    iter++;
  }
  return array[0];
};

File: 7-kyu/simple-fun-#50--array-conversion/python/solution.py
from itertools import islice
from functools import reduce
from operator import mul

def array_conversion(arr, iteration=1):
    if len(arr) == 1:
        return arr[0]
    if iteration % 2 == 1:
        next_arr = [sum(islice(arr, idx, idx + 2)) for idx in range(0, len(arr), 2)]
        return array_conversion(next_arr, iteration + 1)
    else:
        next_arr = [reduce(mul, islice(arr, idx, idx + 2), 1) for idx in range(0, len(arr), 2)]
        return array_conversion(next_arr, iteration + 1)